frontend/src/container/BillingContainer/BillingContainer.tsx

```diff
@@ -371,6 +372,17 @@ export default function BillingContainer(): JSX.Element {
 		</Typography>
 	);
 
+	const handleCsvDownload = useCallback((): void => {
+		const csv = prepareCsvData(apiResponse);
```

Comment:
Downloading files should be in a try-catch block as it's prone to failures.

Reply:
Done, added try-catch with notification (error-success)